src/components/inputs/TextInput.tsx

```diff
@@ -1,24 +1,44 @@
 "use client";
 
-import React, { ChangeEvent, useState } from "react";
-
+import React, { ChangeEvent, useState, useRef, ElementRef } from "react";
+import { XMarkIcon } from "@heroicons/react/24/outline";
 import { EyeSlashIcon, EyeIcon } from "@heroicons/react/24/solid";
+
 import Label from "./Label";
 import FieldMessage from "./FieldMessage";
+import Button from "@/components/Button";
 
 import { cn } from "@/lib/utils";
 
-export interface TextInputProps
+interface CommonTextInputProps
   extends React.InputHTMLAttributes<HTMLInputElement> {
   id: string;
   label?: string;
   placeholder: string;
   suggestion?: string;
   maxLength?: number;
   errorMessage?: string | undefined;
-  type?: string;
 }
 
+type ConditionalTextInputProps =
+  | {
+      inputGroupContent?: JSX.Element;
+      submitButtonText: string;
+      clearInputAction: () => void;
+    }
+  | {
+      inputGroupContent?: JSX.Element;
+      submitButtonText?: undefined;
+      clearInputAction?: never;
+    }
+  | {
+      inputGroupContent?: JSX.Element;
+      submitButtonText?: never;
+      clearInputAction?: undefined;
+    };
```

Comment:
Curious why you made separate types like this instead of just making 1 type and make submitbuttontext and clearinputaction optional? Not sure what the difference is here

Reply:
@Dan-Y-Ko With less props doesn't make much sense anymore. I removed it.